soc/espressif/esp32c6/default.ld

```diff
@@ -0,0 +1,807 @@
+    __bss_start = ABSOLUTE(.);
+    _bss_start = ABSOLUTE(.);
+
+    *libbtdm_app.a:(.bss .bss.* COMMON)
+    . = ALIGN (4);
+    _btdm_bss_end = ABSOLUTE(.);
+
+    *(.dynsbss)
+    *(.sbss)
+    *(.sbss.*)
+    *(.gnu.linkonce.sb.*)
+    *(.scommon)
+    *(.sbss2)
+    *(.sbss2.*)
+    *(.gnu.linkonce.sb2.*)
+    *(.dynbss)
+    *(.bss)
+    *(.bss.*)
+    *(.share.mem)
+    *(.gnu.linkonce.b.*)
+    *(COMMON)
+    . = ALIGN (16);
+    __bss_end = ABSOLUTE(.);
+    _bss_end = ABSOLUTE(.);
+  } GROUP_LINK_IN(RAMABLE_REGION)
+
+  /* Provide total SRAM usage, including IRAM and DRAM */
+  _image_ram_start = _iram_start;
+  #include <zephyr/linker/ram-end.ld>
+
+  ASSERT(((__bss_end - ORIGIN(sram0_0_seg)) <= LENGTH(sram0_0_seg)), "DRAM segment data does not fit.")
+
+  /* --- END OF DRAM --- */
+
+  /* --- START OF .flash.text --- */
+
+  .flash.align_text (NOLOAD):
+  {
+    /* Subsequent segment lma align */
+    . = ALIGN(CACHE_ALIGN);
+  } GROUP_LINK_IN(ROMABLE_REGION)
+
+  /* Symbols used during the application memory mapping */
+  _image_irom_start = LOADADDR(.flash.text);
+  _image_irom_size = SIZEOF(.flash.text);
+  _image_irom_vaddr = ADDR(.flash.text);
+
+  .flash.text : ALIGN(0x10)
+  {
+    _stext = .;
+    _instruction_reserved_start = ABSOLUTE(.);
+    _text_start = ABSOLUTE(.);
+    _instruction_reserved_start = ABSOLUTE(.);
+    __text_region_start = ABSOLUTE(.);
+
+#if !defined(CONFIG_ESP32_WIFI_IRAM_OPT)
+    *libnet80211.a:( .wifi0iram  .wifi0iram.* .wifislpiram .wifislpiram.*)
+    *libpp.a:( .wifi0iram  .wifi0iram.* .wifislpiram .wifislpiram.*)
+#endif /* CONFIG_ESP32_WIFI_IRAM_OPT */
+
+#if !defined(CONFIG_ESP32_WIFI_RX_IRAM_OPT)
+    *libnet80211.a:( .wifirxiram  .wifirxiram.* .wifislprxiram .wifislprxiram.*)
+    *libpp.a:( .wifirxiram  .wifirxiram.* .wifislprxiram .wifislprxiram.*)
+#endif /* CONFIG_ESP32_WIFI_RX_IRAM_OPT */
+
+    *(.literal .text .literal.* .text.*)
+    *(.stub .gnu.warning .gnu.linkonce.literal.* .gnu.linkonce.t.*.literal .gnu.linkonce.t.*)
+    *(.irom0.text) /* catch stray ICACHE_RODATA_ATTR */
+
+    *(.fini.literal)
+    *(.fini)
+
+    *(.gnu.version)
+
+    /** CPU will try to prefetch up to 16 bytes of
+      * of instructions. This means that any configuration (e.g. MMU, PMS) must allow
+      * safe access to up to 16 bytes after the last real instruction, add
+      * dummy bytes to ensure this
+      */
+    . += 16;
+
+    _instruction_reserved_end = ABSOLUTE(.);
+    _text_end = ABSOLUTE(.);
+    _instruction_reserved_end = ABSOLUTE(.);
+    __text_region_end = ABSOLUTE(.);
+    _etext = .;
+
+  } GROUP_DATA_LINK_IN(CACHED_REGION, ROMABLE_REGION)
+
+  /* --- END OF .flash.text --- */
+
+  /* --- START OF .rodata --- */
+
+  /* Align next section to 64k to allow mapping */
+  .flash.align_rodata (NOLOAD) :
+  {
+    /* Subsequent segment lma and vma align */
+    . = ALIGN(CACHE_ALIGN);
+  } GROUP_DATA_LINK_IN(CACHED_REGION, ROMABLE_REGION)
+
+  .flash.align_rom (NOLOAD) :
```

Comment:
I think it needs to account for the .text region:
````c
  .flash.align_rom (NOLOAD) :
  {
    /* Spacer in the IROM address to avoid interfering with the DROM address
     * because DROM and IROM regions share the same address space */
    . += SIZEOF(.flash.text);
    . = ALIGN(CACHE_ALIGN);
  } GROUP_LINK_IN(CACHEd_REGION)
  ```

Reply:
@marekmatej PTAL as well.

Reply:
please, update to
```
  /* Align next section to 64k to allow mapping */
  .flash.align_rom (NOLOAD) :
  {
    . = ALIGN(CACHE_ALIGN);
  } GROUP_LINK_IN(ROMABLE_REGION)

  .flash.align_rodata (NOLOAD) :
  {
    . = ALIGN(CACHE_ALIGN);
  } GROUP_LINK_IN(CACHED_REGION)
```